packages/expo/cli/start/server/__tests__/openPlatforms-test.ts

```diff
@@ -0,0 +1,98 @@
+import { AbortCommandError } from '../../../utils/errors';
+import { DevServerManager } from '../DevServerManager';
+import { openPlatformsAsync } from '../openPlatforms';
+
+const asMock = <T extends (...args: any[]) => any>(fn: T): jest.MockedFunction<T> =>
+  fn as jest.MockedFunction<T>;
+
+function createDevServerManager() {
+  return {
+    getDefaultDevServer: jest.fn(() => ({
+      openPlatformAsync: jest.fn(),
+    })),
+    getWebDevServer: jest.fn(() => ({
+      openPlatformAsync: jest.fn(),
+    })),
+    ensureWebDevServerRunningAsync: jest.fn(() => Promise.resolve()),
+  } as unknown as DevServerManager;
+}
+
+it(`opens all platforms`, async () => {
+  const options = {
+    android: true,
+    ios: true,
+    web: true,
+  };
+
+  const manager = createDevServerManager();
+  const openedNative = await openPlatformsAsync(manager, options);
+  expect(openedNative).toBe(true);
+
+  expect(manager.getDefaultDevServer).toHaveBeenCalledTimes(2);
+  expect(manager.getWebDevServer).toHaveBeenCalledTimes(1);
+  expect(manager.ensureWebDevServerRunningAsync).toHaveBeenCalledTimes(1);
+});
+
+it(`opens web only`, async () => {
+  const options = {
+    android: false,
+    ios: false,
+    web: true,
+  };
+  const manager = createDevServerManager();
+
+  const openedNative = await openPlatformsAsync(manager, options);
+  expect(openedNative).toBe(false);
+
+  expect(manager.getDefaultDevServer).toHaveBeenCalledTimes(0);
+  expect(manager.getWebDevServer).toHaveBeenCalledTimes(1);
+  expect(manager.ensureWebDevServerRunningAsync).toHaveBeenCalledTimes(1);
+});
+
+it(`rethrows assertions`, async () => {
+  const manager = createDevServerManager();
+  asMock(manager.getDefaultDevServer).mockImplementationOnce(
+    () =>
+      ({
+        openPlatformAsync() {
+          throw new Error('Failed');
+        },
+      } as any)
+  );
+  const options = {
+    android: true,
+    ios: true,
+    web: true,
+  };
+  await expect(openPlatformsAsync(manager, options)).rejects.toThrow(/Failed/);
+
+  expect(manager.getDefaultDevServer).toHaveBeenCalledTimes(1);
+  expect(manager.getWebDevServer).toHaveBeenCalledTimes(0);
+  expect(manager.ensureWebDevServerRunningAsync).toHaveBeenCalledTimes(0);
+});
+
+it(`surfaces aborting`, async () => {
+  const manager = createDevServerManager();
+  asMock(manager.getDefaultDevServer)
+    .mockReturnValueOnce({
+      openPlatformAsync() {},
+    } as any)
+    .mockImplementationOnce(
+      () =>
+        ({
+          openPlatformAsync() {
+            throw new AbortCommandError();
+          },
```

Comment:
These mock functions, specifically the second one, need to be async. They are currently masking a bug (a comment on that in openPlatforms.ts).

What's happening is that the second mock openPlatformAsync method here immediately throws an AbortCommandError before Promise.allSettled is called. The code to filter through the results and check for `error.code === 'ABORTED'` doesn't run at all. Because this test looks for an AbortCommandError instance, the test passes, but it's receiving the wrong instance of AbortCommandError.

Making the mock function async ensures it returns a rejected promise instead of throwing.

```suggestion
      async openPlatformAsync() {},
    } as any)
    .mockImplementationOnce(
      () =>
        ({
          async openPlatformAsync() {
            throw new AbortCommandError();
          },
```